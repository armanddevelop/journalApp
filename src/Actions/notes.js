import { db } from "../FireBase/fireBaseConfig";
import { types } from "../Types/types";

export const notesActiveAction = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
});

export const startNewNoteAction = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const newNote = {
      title: "",
      body: "",
      date: new Date().getTime(),
    };
    const docRef = await db.collection(`${uid}/jornal/notes`).add(newNote);
    dispatch(notesActiveAction(docRef.id, newNote));
  };
};
