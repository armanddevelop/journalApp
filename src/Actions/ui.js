import { types } from "../Types/types";

export const setErrorAction = (error) => ({
  type: types.uiSetError,
  payload: error,
});

export const unSetErrorAction = () => ({
  type: types.uiRemoveError,
});

export const startLoading = () => ({
  type: types.uiStartLoading,
});

export const finishLoading = () => ({
  type: types.uiFinishLoading,
});
