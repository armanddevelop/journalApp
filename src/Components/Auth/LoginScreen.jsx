import { Link, useLocation } from "react-router-dom";
import { useForm } from "../../Hooks/useForm";
import { Form } from "../Form/Form";
import { Google } from "../Social/Google";
import { identifierPage } from "../Utils/identifierPage";

const initialState = {
  email: "alicha@test.com",
  password: "123456",
};
export const LoginScreen = () => {
  const pageName = "Login";
  const location = useLocation();
  const page = identifierPage(location, pageName);
  const { values, handleInputChange } = useForm(initialState);
  return (
    <>
      <h3 className="auth__title">{pageName}</h3>
      <Form
        pageName={pageName}
        page={page}
        values={values}
        handleInputChange={handleInputChange}
      />
      <Google />
      <Link className="link" to="/auth/register">
        Create new account
      </Link>
    </>
  );
};
