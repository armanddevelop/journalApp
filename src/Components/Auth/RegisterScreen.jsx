import { Link, useLocation } from "react-router-dom";
import { useForm } from "../../Hooks/useForm";
import { Form } from "../Form/Form";
import { identifierPage } from "../Utils/identifierPage";
const initialState = {
  name: "alicha",
  email: "alicha@test.com",
  password: "123456",
  confirmPassword: "123456",
};

export const RegisterScreen = () => {
  const pageName = "Register";
  const location = useLocation();
  const page = identifierPage(location, pageName);
  const { values, handleInputChange } = useForm(initialState);
  return (
    <div>
      <h3 className="auth__title">{pageName}</h3>
      <Form
        pageName={pageName}
        page={page}
        values={values}
        handleInputChange={handleInputChange}
      />
      <Link className="link" to="/auth/login">
        Already Register?
      </Link>
    </div>
  );
};
