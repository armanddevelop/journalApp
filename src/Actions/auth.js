import { types } from "../Types/types";
import { firebase, googleAuthProvider } from "../FireBase/fireBaseConfig";
import {
  finishLoading,
  startLoading,
  errorInFirebase,
  removeErrorInFirebase,
} from "./ui";
import { notesLogOutAction } from "./notes";

export const logInAction = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const logOutAction = () => ({
  type: types.logout,
});

export const startLogOut = () => {
  return async (dispatch) => {
    try {
      await firebase.auth().signOut();
      dispatch(logOutAction());
      dispatch(notesLogOutAction());
    } catch (error) {
      console.log("shit happen in startLogOut ", error);
    }
  };
};

export const getGoogleLogInAction = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
      });
  };
};

export const getLoginUserAction = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};

export const registerUserAction = (name, email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};
