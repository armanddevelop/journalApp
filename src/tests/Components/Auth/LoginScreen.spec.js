import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { getGoogleLogInAction } from "../../../Actions/auth";
import { removeErrorInFirebase, unSetErrorAction } from "../../../Actions/ui";
import { LoginScreen } from "../../../Components/Auth/LoginScreen";

const mockLocation = jest.fn();
const mockidentifierPage = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => mockLocation,
}));

jest.mock("../../../Components/Utils/identifierPage", () => ({
  identifierPage: () => mockidentifierPage,
}));

jest.mock("../../../Actions/auth", () => ({
  getGoogleLogInAction: jest.fn(),
}));

jest.mock("../../../Actions/ui", () => ({
  removeErrorInFirebase: jest.fn(),
  unSetErrorAction: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
  auth: {},
  ui: {
    errorMsgFb: {
      error: false,
      msgError: "",
    },
  },
};
let store = mockStore(initState);
store.dispatch = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <LoginScreen />
    </MemoryRouter>
  </Provider>
);
describe("Test in LoginScreen component", () => {
  beforeEach(() => {
    store = mockStore(initState);
    jest.clearAllMocks();
  });

  it("snapShot to LoginScreen component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Login with google should fire the google login", () => {
    wrapper.find(".google-btn").prop("onClick")();

    expect(getGoogleLogInAction).toHaveBeenCalled();
  });

  it("removeErrorInFirebase and unSetErrorAction should been called", () => {
    wrapper.find(".link").first().prop("onClick")();

    expect(removeErrorInFirebase).toHaveBeenCalled();
    expect(unSetErrorAction).toHaveBeenCalled();
  });
});
