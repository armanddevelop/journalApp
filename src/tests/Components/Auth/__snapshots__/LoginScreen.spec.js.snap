// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in LoginScreen component snapShot to LoginScreen component 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Login
  </h3>,
  <form
    className="animate__animated animate__fadeIn"
    onSubmit={[Function]}
  >
    Array [
      <input
        autoComplete="off"
        className="auth__input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />,
      <input
        autoComplete="off"
        className="auth__input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="alicha@test.com"
      />,
      <input
        className="auth__input"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value="123456"
      />,
      <input
        className="auth__input"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
      />,
    ]
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Login
    </button>
  </form>,
  <div
    className="auth__social_networks animate__animated animate__fadeIn"
  >
    <p>
      Login with Social Networks
    </p>
    <div
      className="google-btn"
      onClick={[Function]}
    >
      <div
        className="google-icon-wrapper"
      >
        <img
          alt="google button"
          className="google-icon"
          src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
        />
      </div>
      <p
        className="btn-text"
      >
        <span>
          Sign in with google
        </span>
      </p>
    </div>
  </div>,
  <a
    className="link"
    href="/auth/register"
    onClick={[Function]}
  >
    Create new account
  </a>,
]
`;
